camayoc/tests/qpc/api/v1/sources/test_network_sources.py

```diff
@@ -213,3 +213,76 @@ def test_negative_update_invalid(
     assert_source_update_fails(original_data, src)
     src.hosts = ['1**2@33^']
     assert_source_update_fails(original_data, src)
+
+
+def test_create_empty_str_host_valid(shared_client, cleanup):
+    """Create a host manager source and then add host data with empty string.
+
+    :id: 4BFBF5B0-81BD-48A3-8C57-BAE55590285D
+    :description: Create network source of multiple hosts including an empty
+        string and credential.
+    :steps:
+        1) Create a network credential
+        2) Create a source with multiple host including empty string
+    :expectedresults: New source is created with all hosts except the empty
+        string filtered out.
+    """
+    EMPTY_STR_HOST_DATA = ['192.0.2.1', '192.0.2.10', '']
+    # initialize & create original credential & source
+    pwd_cred = Credential(
+        cred_type=NETWORK_TYPE,
+        client=shared_client,
+        password=uuid4()
+    )
+    pwd_cred.create()
+
+    src = Source(
+        source_type=NETWORK_TYPE,
+        client=shared_client,
+        hosts=EMPTY_STR_HOST_DATA,
+        credential_ids=[pwd_cred._id],
+    )
+    src.create()
+
+    # add the ids to the lists to destroy after the test is done
+    cleanup.extend([pwd_cred, src])
+
+
+def test_update_empty_str_host_valid(shared_client, cleanup):
+    """Create a host manager source and then add host data with empty string.
+
+    :id: 850B06BA-8E3F-4699-B24B-E75AAD20A63A
+    :description: Assert that we can update a source with hosts including an
+         empty string, but the empty string is filtered out.
+    :steps:
+        1) Create a valid network credential and source
+        2) Attempt to update with host list including empty string
+        3) Assert update is made but with the empty string filtered out.
+    :expectedresults: Source is updated with only valid host list.
+    """
+    EMPTY_STR_HOST_DATA = ['192.0.2.1', '192.0.2.10', '']
+    # initialize & create original credential & source
+    pwd_cred = Credential(
+        cred_type=NETWORK_TYPE,
+        client=shared_client,
+        password=uuid4()
+    )
+    pwd_cred.create()
+
+    src = Source(
+        source_type=NETWORK_TYPE,
+        client=shared_client,
+        hosts=['127.0.0.1'],
+        credential_ids=[pwd_cred._id],
+    )
+    src.create()
+
+    # add the ids to the lists to destroy after the test is done
+    cleanup.extend([pwd_cred, src])
+
+    src.hosts = EMPTY_STR_HOST_DATA
+    EMPTY_STR_HOST_DATA.remove('')
+    # now that we took out the empty string rename for readability's sake
+    HOSTS_WITHOUT_EMPTY_STR = EMPTY_STR_HOST_DATA
+    updated_data = src.update().json()
```

Comment:
You should call update right after the `src.hosts` being updated, calling here you are actually passing a list without the empty string.  See below for an example:

```
In [1]: l = ['1', '2', '']

In [2]: renamed = l

In [3]: l.remove('')

In [4]: l
Out[4]: ['1', '2']
```